docs/how-tos/working-with-postgresql-data-types.rst

```diff
@@ -0,0 +1,153 @@
+.. _working_with_types:
+
+Working with PostgreSQL data types
+==================================
+
+You may be wondering if PostgREST allows the use of special literal values, such as ``now`` for timestamps, ``yes`` for booleans or time values including the time zone. The answer is yes to all of them, that is because PostgREST makes use of PostgreSQL text representation to work with data types.
+
+Timestamps
+----------
+
+You can use the **time zone** to filter or send data if needed. Let's use this table as an example:
+
+.. code-block:: postgres
+
+  create table reports (
+    id int primary key
+    , due_date timestamptz
+  );
+
+Suppose you are located in Sydney and want create a report with the date in the local time zone. Your request should look like this:
+
+.. tabs::
+
+  .. code-tab:: http
+
+    POST /reports HTTP/1.1
+    Content-Type: application/json
+
+    [{ "id": 1, "due_date": "2022-02-24 11:10:15 Australia/Sydney" },
+     { "id": 2, "due_date": "2022-02-27 22:00:00 Australia/Sydney" }]
+
+  .. code-tab:: bash Curl
+
+    curl "http://localhost:3000/reports" \
+      -X POST -H "Content-Type: application/json" \
+      -d '[{ "id": 1, "due_date": "2022-02-24 11:10:15 Australia/Sydney" },{ "id": 2, "due_date": "2022-02-27 22:00:00 Australia/Sydney" }]'
+
+Someone located in Cairo can retrieve the data using their local time, too:
+
+.. tabs::
+
+  .. code-tab:: http
+
+    GET /reports?due_date=eq.2022-02-24+02:10:15+Africa/Cairo HTTP/1.1
+
+  .. code-tab:: bash Curl
+
+    curl "http://localhost:3000/reports?due_date=eq.2022-02-24+02:10:15+Africa/Cairo"
+
+.. code-block:: json
+
+  [
+    {
+      "id": 1,
+      "due_date": "2022-02-23T19:10:15-05:00"
+    }
+  ]
+
+The response has the date in the time zone configured by the server: ``UTC -05:00``.
+
+You can use other comparative filters and also `PostgreSQL special date/time input values <https://www.postgresql.org/docs/current/datatype-datetime.html#DATATYPE-DATETIME-SPECIAL-TABLE>`_. For instance, to get the reports that are due after today you would do:
+
+.. tabs::
+
+  .. code-tab:: http
+
+    GET /reports?due_date=gt.today HTTP/1.1
+
+  .. code-tab:: bash Curl
+
+    curl "http://localhost:3000/reports?due_date=gt.today"
+
+.. code-block:: json
+
+  [
+    {
+      "id": 2,
+      "due_date": "2022-02-27T06:00:00-05:00"
+    }
+  ]
+
+But what if, for any reason, you need to cast the ``today`` special value to a timezone different from the server. That would mean using the ``AT TIME ZONE`` construct, which can't be done directly in PostgREST, so your best bet would be to create a view, a :ref:`function <s_procs>` or :ref:`computed columns <computed_cols>`. Let's use the last one for the next example:
```

Comment:
I'd say that's a limitation of the literal value in PostgreSQL, so not directly related to PostgREST. So this would be better phrased as a limitation of the text representation.

Reply:
Btw, I've just tested this and it's not true, I can do:

```
select 'today Australia/Sydney'::timestamptz;

      timestamptz
------------------------
 2022-03-02 08:00:00-05
```

It is the `now` literal the one that doesn't work with a time zone.

Reply:
> Btw, I've just tested this and it's not true

You're right about the `timestamptz`. I also made a mistake: `due_date` must be of type `timestamp` without time zone.

Thinking about it I'm starting to doubt this use case, due to the fact that the dates are always saved in UTC in the database, there shouldn't be a need to cast `now` to a time zone unless the values inserted in the table are in that same time zone (implying the data was saved incorrectly). Maybe there's a better use case to use as an example for this?

Reply:
> implying the data was saved incorrectly

Yeah, I also think is a weird use case this one. How about just leaving this part out then? Then the timestamp section will be short and sweet.